### PR TITLE
Update cache tests to enable the java to .net and python tests

### PR DESCRIPTION
**Pull request by henrik-me**
A few issues:
* Maven was not found as the process is a cmd, 
* The process runner was not able to run the cmd with-out a complete path to the cmd. 
* The latest Java library had introduced some updates to the public api
* Python executable location improved to enable local test runs (should not be needed but was not working locally on my box)

TODO:
* Lookup env var M2_HOME to find maven installation
* Add logging to the Java cache test

### CODE REVIEW

**Comment by henrik-me** [comment]
Verified on a release build and the cache tests passes, however one integration test fails:
SeleniumGetAuthCode_RedeemForAt_CommonAuthority_Async

File: tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
         <dependency>
             <groupId>com.microsoft.azure</groupId>
             <artifactId>msal4j</artifactId>
-            <version>0.5.0-preview</version>
+            <version>0.6.0-preview</version>
```
Comment (bgavrilMS) [inline tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml]: We should look into version ranges / using LATEST, but as a separate work item, as it does not seem to be trivial.

**Comment by bgavrilMS** [comment]
I’ve made some changes to harden that test, the stuff done here has no impact.

**Comment by bgavrilMS** [comment]
Tests are passing, this can be checked in.

**Comment by henrik-me** [comment]
> Tests are passing, this can be checked in.

I will be optimizing the code looking for Maven and if tests passes I will merge. thanks

**Pull request merged**